pkg/reconciler/namespace/namespace.go

```diff
@@ -262,3 +268,41 @@ func (r *Reconciler) reconcileBroker(ctx context.Context, ns *corev1.Namespace)
 	// Don't update anything that is already present.
 	return current, nil
 }
+
+// cleanup of unwanted brokers, service account and rolebinding in namespace with no injection label.
+func (r *Reconciler) cleanupResources(ctx context.Context, ns string) error {
+	// delete broker
+	err := r.EventingClientSet.EventingV1alpha1().Brokers(ns).DeleteCollection(
+		&metav1.DeleteOptions{},
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", resources.ResourceInjectionLabel, "true"),
+			FieldSelector: fmt.Sprintf("metadata.name=%s", resources.DefaultBrokerName),
+		})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		logging.FromContext(ctx).Error("Error deleting broker in namespace", zap.String("brokername", resources.DefaultBrokerName), zap.Error(err))
```

Comment:
it is an error? perhaps the user deleted it and disabled the namespace annotation?